### ISSUE TITLE
Error: unrecognized attach_file error in getThreadHistory

### ISSUE THREAD

**Issue by allesmi** [Schmavery/facebook-chat-api#404]
I wrote a program that saves the history of a group chat spanning several months.
On some messages that are older than Oct 8th the following exception was raised:

```
ERR! Error in getUserInfo Error: unrecognized attach_file error
ERR! Error in getUserInfo     at _formatAttachment (myproject/node_modules/facebook-chat-api/utils.js:331:13)
ERR! Error in getUserInfo     at myproject/node_modules/facebook-chat-api/utils.js:340:14
ERR! Error in getUserInfo     at Array.map (native)
ERR! Error in getUserInfo     at formatAttachment (myproject/node_modules/facebook-chat-api/utils.js:337:36)
ERR! Error in getUserInfo     at formatMessage (myproject/node_modules/facebook-chat-api/utils.js:359:18)
ERR! Error in getUserInfo     at Array.map (native)
ERR! Error in getUserInfo     at myproject/node_modules/facebook-chat-api/src/getThreadHistory.js:54:52
ERR! Error in getUserInfo     at myproject/node_modules/facebook-chat-api/src/getUserInfo.js:49:14
ERR! Error in getUserInfo     at tryCatcher (myproject/node_modules/facebook-chat-api/node_modules/bluebird/js/main/util.js:26:23)
ERR! Error in getUserInfo     at Promise._settlePromiseFromHandler (myproject/node_modules/facebook-chat-api/node_modules/bluebird/js/main/promise.js:510:31)
ERR! Error in getUserInfo     at Promise._settlePromiseAt (myproject/node_modules/facebook-chat-api/node_modules/bluebird/js/main/promise.js:584:18)
ERR! Error in getUserInfo     at Promise._settlePromises (myproject/node_modules/facebook-chat-api/node_modules/bluebird/js/main/promise.js:700:14)
ERR! Error in getUserInfo     at Async._drainQueue (myproject/node_modules/facebook-chat-api/node_modules/bluebird/js/main/async.js:123:16)
ERR! Error in getUserInfo     at Async._drainQueues (myproject/node_modules/facebook-chat-api/node_modules/bluebird/js/main/async.js:133:10)
ERR! Error in getUserInfo     at Immediate.Async.drainQueues [as _onImmediate] (myproject/node_modules/facebook-chat-api/node_modules/bluebird/js/main/async.js:15:14)
ERR! Error in getUserInfo     at processImmediate [as _immediateCallback] (timers.js:396:17)
ERR! Error in getUserInfo  [Error: unrecognized attach_file error]
```

Upon further inspection it appears that the attachment object has no other properties than `"type": "error"`. As a quick workaround I return an empty object in the case of "error".

While this appears to be an issue at the Facebook server side, what would be an elegant way to catch this error in the API?

**Comment by Schmavery** [comment]
I believe we map over all the returned messages [here](https://github.com/Schmavery/facebook-chat-api/blob/master/src/getThreadHistory.js#L56).  You could strip the error out at this stage.

If you scroll up to that point in the chat in your browser, does anything appear where the error is?  If facebook renders something, it might be worth returning that value for completeness... otherwise I totally agree it would be nice to strip it out.

**Comment by allesmi** [comment]
This is how this message is rendered:
![fbchaterror](https://cloud.githubusercontent.com/assets/3799846/23105289/47dcd68e-f6dd-11e6-9223-f5e6df3c0dbc.png)

The corresponding message object, as returned by the api and without personal information:
```
{ type: 'message',
  senderName: 'user name',
  senderID: 'fbid:xxx',
  participantNames: [ 'user' ],
  participantIDs: [ 'fbid:xxx' ],
  body: 'message body',
  threadID: 'xxx',
  threadName: 'user name',
  location: null,
  messageID: 'mid.xxx:xxx',
  attachments: [ { type: 'error', ID: '', image_data: {} } ],
  timestamp: 1468880844715,
  timestampAbsolute: 'July 19, 2016',
  timestampRelative: '7/19/16',
  timestampDatetime: 'July 19, 2016 12:27 am' }

```

Two objects got passed to the `_formatAttachment` function: `attachment1 = {"attach_type": "error}` and `attachment2 = {"id": "", "image_data": {} }`

I cannot remember what the message originally contained, I am sure I would have noticed if it has been an error. Given the context in this thread it was most likely a YouTube url.

**Comment by Schmavery** [comment]
Hmm.. thanks for hunting this down.   What's your opinion here?  I'm tempted to say that by stripping out this error we actually lose information that the desktop client gets.  As long as this error is returned in a clean, documented way, I'd probably prefer to keep it in, but I could be convinced otherwise.

**Comment by allesmi** [comment]
After investigating further I found out I made some flawed statements.

1. `attachment2` is undefined, the object with properties `id` and `image_data` is created within the function `_formatAttachment`.
2. I was using `facebook-chat-api` version 1.0.8 in my project. The most recent version 1.3.0 already respects messages of type `error`, returning both attachments unchanged. I consider this to be the sanest behaviour. In my particular use case there was only a single error message among hundreds of valid ones.
